query/common/non_aggregate_query_result.go

```diff
@@ -0,0 +1,50 @@
+//  Copyright (c) 2017-2018 Uber Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package common
+
+import "github.com/uber/aresdb/utils"
+
+const MatrixDataKey = "matrixData"
+const HeadersKey = "headers"
+
+// AQLNonAggregatedQueryResult represents non-aggregated query result
+// - there will be a "headers" key, value will be a list of column names
+// - there will be a "matrixData" key, value will be a 2d arary of values (row formated)
+type AQLNonAggregatedQueryResult map[string]interface{}
+
+// Append appends single value to specific row, result rows must be built in order
+func (r AQLNonAggregatedQueryResult) Append(row int, value interface{}) error {
```

Comment:
where to use this new type to return to caller? how is the original aqltimeseriesresult?

Reply:
supposedly in PostProcess and handler. for non-agg query use AQLNonAggregatedQueryResult other wise use time-series-result. we might need to use `map[string]interface{}` in those places and cast these 2 types to map.
another option is to squash everything in the same general result object. wdyt?

Reply:
merged the methods to 1 struct and renamed to `AQLQueryResult`